src/config/paths-to-module-name-mapper.ts

```diff
@@ -47,12 +46,21 @@ export const pathsToModuleNameMapper = (
 
         return `${enrichedPrefix}${enrichedTarget.replace(/\*/g, '$1')}`
       })
-      pattern = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}$`
+
+      if (removeJsExtension) {
+        const patternWithJs = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}\\.js$`
+        jestMap[patternWithJs] = paths.length === 1 ? paths[0] : paths
+      }
+      const pattern = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}$`
```

Comment:
This looks a bit strange to me that we have an if here which adds extra thing to jestMap but what is the else case?